cirq/google/engine/engine.py

```diff
@@ -405,65 +340,145 @@ def run_sweep(
+                for sweep in sweeps
+            ]
+            return context_dict
+        elif proto_version == ProtoVersion.V2:
+            run_context = v2.run_context_pb2.RunContext()
+            for sweep in sweeps:
+                sweep_proto = run_context.parameter_sweeps.add()
+                sweep_proto.repetitions = repetitions
+                api_v2.sweep_to_proto(sweep, out=sweep_proto.sweep)
+
+            return _any_dict_from_msg(run_context)
+        else:
+            raise ValueError(
+                'invalid run context proto version: {}'.format(proto_version))
+
+    def create_program(self,
+                       program: Program,
+                       program_id: Optional[str] = None,
+                       gate_set: Optional[SerializableGateSet] = gate_sets.XMON
+                      ) -> 'EngineProgram':
+        """Wraps a Circuit or Scheduler for use with the Quantum Engine.
+
+        Args:
+            program: The Circuit or Schedule to execute. If a circuit is
+                provided, a moment by moment schedule will be used.
+            program_id: A user-provided identifier for the program. This must be
+                unique within the Google Cloud project being used. If this
+                parameter is not provided, a random id of the format
+                'prog-######' will be generated.
+            gate_set: The gate set used to serialize the circuit. The gate set
+                must be supported by the selected processor
+        """
+        if not program_id:
+            program_id = _make_random_id('prog-')
+
+        parent_name = 'projects/%s' % self.project_id
+        program_name = '%s/programs/%s' % (parent_name, program_id)
+        # Create program.
+        request = {
+            'name': program_name,
+            'code': self._serialize_program(program, gate_set),
+        }
+        result = self.service.projects().programs().create(
+            parent=parent_name, body=request).execute()
+
+        return EngineProgram(result['name'], self)
+
+    def _serialize_program(
+            self,
+            program: Program,
+            gate_set: Optional[SerializableGateSet] = gate_sets.XMON
+    ) -> Dict[str, Any]:
+        if self.proto_version == ProtoVersion.V1:
+            schedule = self.program_as_schedule(program)
+            schedule.device.validate_schedule(schedule)
+
+            program_descriptor = v1.program_pb2.Program.DESCRIPTOR
+            program_dict = {}  # type: Dict[str, Any]
+            program_dict['@type'] = TYPE_PREFIX + program_descriptor.full_name
+            program_dict['operations'] = [
+                op for op in api_v1.schedule_to_proto_dicts(schedule)
+            ]
+            return program_dict
+        elif self.proto_version == ProtoVersion.V2:
+            if isinstance(program, Schedule):
+                program.device.validate_schedule(program)
+            program = gate_set.serialize(program)
```

Comment:
need to assert that gate_set is not None.  May need to cast it to SerializableGateSet